### ISSUE TITLE
[Question] Playwright Component Testing and Service Worker

### ISSUE THREAD

**Issue by mvill** [microsoft/playwright#20659]
Hi,

I'm trying to make work together Playwright Component Testing and Mock Service Worker (MSW) but I'm facing an issue :
"Service Worker registration blocked by Playwright"

Even if i add the option "serviceWorkers: allow" in config, the problem persists.

By diving in the code, I came across this :
https://github.com/microsoft/playwright/blob/main/packages/playwright-test/src/mount.ts#L36

"serviceWorkers: 'block'" is hard coded and there is no way to change it.

Is there a reason for this ?

**Comment by mxschmitt** [comment]
What is your use-case for MSW? Usually you use Playwright's route interception for manipulating network, see here: https://playwright.dev/docs/network#modify-requests

**Comment by mxschmitt** [comment]
Closing for now since it seems stale and answered. Please re-file if you still have questions or issues with Playwright.

**Comment by iulspop** [comment]
Hello, I also wonder why my MSW configuration doesn't intercept requests as expected. I have a React SPA which makes a request to GitHub I want to mock. I configured MSW to start in the client JS code before rendering in React (not in the Playwright Node context by mistake). When I run it in my own browser it works perfectly, but when I run it in Playwright's browser it seems to start but doesn't intercept requests as expected.

**Comment by Crismon96** [comment]
@mxschmitt First of: thanks for working on playwright-ct. I thinks it a great supplementation to the JS testing landscape. 

1. Service workers influence component specific behavior so it would be nice to have them supported somehow in playwright-ct. I would like to test individual components working together without having whole playwright E2E tests.
2. I am using MSW too and it would be much more convenient for me to just use this one layer of mocking than to pass everything through playwright. Using MSW works for playwright E2E tests quite conveniently.

I will follow your recommendation for now and manipulate the network via playwright itself. Do you think it is still worth bringing forward to support service workers in playwright-ct anytime soon?

**Comment by Crismon96** [comment]
For anyone who really doesnt want to copy and maintain a second layer of abstraction this workaround might be handy.
I intercept any requests to my mocked server and pass the body and header data to a function that makes a server side request after the mock server of MSW is spun up.

```
import { expect, test } from '@playwright/experimental-ct-react';
import 'whatwg-fetch';
import { handlers } from '../../../mocks/handlers';
import Assets from '../Features/Assets/Assets';
import { IntegrationWrapper } from './test-utils';

import { setupServer } from 'msw/node';

test.beforeAll(() => {
  const server = setupServer(...handlers);
  server.listen();
});

test.only('should render Assets', async ({ mount, page }) => {
  const makeFetchToMSW = async (
    postData: string | null,
    headers: {
      [key: string]: string;
    },
  ) => {
    const response = await fetch('https://flyby-router-demo.herokuapp.com/', {
      body: postData,
      method: 'POST',
      headers,
    });

    return response;
  };

  await page.route('https://flyby-router-demo.herokuapp.com/', async (route) => {
    const mswResponse = await makeFetchToMSW(route.request().postData(), route.request().headers());

    return route.fulfill({
      status: mswResponse.status,
      body: await mswResponse.text(),
    });
  });

  const component = await mount(
    <IntegrationWrapper>
      <Assets />
    </IntegrationWrapper>,
  );

  await expect(component).toContainText('Learn React');
});
```

I think for MSW this feels like a good enough solution since their server API provides the same utilities as their serviceWorker browser API. 
It would still be cool to see serviceWorker get a better support : )
